Sources/ProjectDescription/Target.swift

```diff
@@ -109,7 +110,62 @@ public struct Target: Codable, Equatable {
                 settings: Settings? = nil,
                 coreDataModels: [CoreDataModel] = [],
                 environment: [String: String] = [:],
-                launchArguments: [String: Bool] = [:])
+                launchArguments: [String: Bool])
```

Comment:
Thanks for keeping backwards compatibility in mind 👍 

A suggestion here to help reduce / simplify this - we could leverage `ExpressibleByDictionaryLiteral` and `ExpressibleByArrayLiteral`.

e.g. We can add another type `LaunchArguments` _(plural)_

```swift
struct LaunchArguments {
    var arguments: [LaunchArgument]
}

extension LaunchArguments: ExpressibleByArrayLiteral {
    typealias ArrayLiteralElement = LaunchArgument
    init(arrayLiteral elements: LaunchArgument...) {
        arguments = elements
    }
}

extension LaunchArguments: ExpressibleByDictionaryLiteral {
    typealias Key = String
    typealias Value = Bool
    init(dictionaryLiteral elements: (String, Bool)...) {
        arguments = elements
            .map { LaunchArgument(name: $0.0, isEnabled: $0.1) }
            .sorted(by: { $0.name < $1.name })
    }
}
```

This would allow us to keep a single `init` with `launchArguments: LaunchArguments` and support both the previous dictionary use case as well as the new array use case. 

The one caveat with my suggestion is, it's backward compatible with literals used directly but any used indirectly will break :(

e.g.

Compatible:

```swift
Target(
   // ...
   launchArguments: ["traceEnabled": false, "uiTesting": true]   
)
```

Not compatible :/

```swift
let arguments = ["traceEnabled": false, "uiTesting": true]
Target(
   // ...
   launchArguments: arguments   
)
```

can be fixed by adding an explicit type:

```swift
let arguments: LaunchArguments = ["traceEnabled": false, "uiTesting": true]
```

Reply:
I was also thinking, what about marking the dictionary init deprecated and later removing it? I think that as long as the order matters, it should also matter to user...or is still the `ExpressibleByDictionaryLiteral` preferred?

Reply:
As suggested by @pepibumur those inits were marked as deprecated 🙂